cmd/wallet.go

```diff
@@ -108,10 +108,23 @@ sure the device is connected, unlocked, and the Spacemesh app is open.`,
 		}
 
 		fmt.Print("Enter a secure password used to encrypt the wallet file (optional but strongly recommended): ")
-		password, err := password.Read(os.Stdin)
+		pass, err := password.Read(os.Stdin)
 		fmt.Println()
 		cobra.CheckErr(err)
-		wk := wallet.NewKey(wallet.WithRandomSalt(), wallet.WithPbkdf2Password([]byte(password)))
+
+		if pass != "" {
+			fmt.Print("Confirm your password: ")
+			confirm, err := password.Read(os.Stdin)
+			fmt.Println()
+			cobra.CheckErr(err)
+
+			if pass != confirm {
+				_, _ = fmt.Fprintln(os.Stderr, "Error: Passwords do not match.")
```

Comment:
let's use `log.Fatalln` instead, as below.

Reply:
Good catch. Fixed! I was trying to imitate `cobra.CheckErr`. I should have read further. :facepalm: